internal/contentcache/contentcache.go

```diff
@@ -39,29 +45,59 @@ type ContentCache struct {
 	mtimeClock timeutil.Clock
 }
 
+// RecoverFileFromCache recovers a file from the cache via metadata
+func (c *ContentCache) RecoverFileFromCache(metadataFile fs.FileInfo) {
+	debug := logger.NewDebug("recover file from cache: ")
+	// validate not a directory and matches gcsfuse pattern
+	if !metadataFile.IsDir() && matchPattern(metadataFile.Name()) {
```

Comment:
nit: This function is only executed when these 2 conditions are true. A better way to express this is:
```go
if maybeMetadataFile.IsDir() {
  return
}
if !isMetadataFileName(maybeMetadataFile.Name()) {
  return
}
// recover the metadata file ...
```